### PR TITLE
r? ianbicking Fixes #309 If a shot has no clips, 404.

### LINKED ISSUE #309: Shot should get deleted when last clip is deleted
After #308 is implemented:

In `lib/shooter.js` the ShotContext should notice when it is destroying that its shot has no clips, and delete the shot on the server at that point.

I don't think we need to be too aggressive in deleting the shot early.  The server could also check if a shot has no clips, and refuse to render it in that case.  (Encourages a dirty database though.)

### PR DESCRIPTION
**Pull request by fzzzy**

### CODE REVIEW

File: server/src/server.js [hunk 1]
```diff
@@ -175,7 +175,7 @@ app.get("/", function (req, res) {
 app.get("/:id/:domain", function (req, res) {
   let shotId = req.params.id + "/" + req.params.domain;
   Shot.get(req.backend, shotId).then((shot) => {
-    if (! shot) {
+    if (! shot || Object.getOwnPropertyNames(shot._clips).length === 0) {
```
Comment (ianb) [inline server/src/server.js]: `shot.clipNames().length === 0` would be a better way to say this

Comment (fzzzy) [inline server/src/server.js]: We only have the json at this point, don't we?

Comment (ianb) [inline server/src/server.js]: No, it's a real instance (created by `Shot.get()`).  Also if it was JSON then it would be `shot.clips` and not `shot._clips`

Comment (fzzzy) [inline server/src/server.js]: Ok cool. I misunderstood while inspecting the object

Comment (fzzzy) [inline server/src/server.js]: Fixed.

**Pull request merged**